BetterPasswordPlugin.inc.php

```diff
@@ -160,24 +160,52 @@ function checkPassword($hookName, $args) {
 		if ($this->getSetting(CONTEXT_SITE, 'betterPasswordCheckBlacklist')) {
 			$badPassword = false;
 			$lowerPassword = strtolower($password);
-			foreach ($this->getBlacklists() as $filename) {
-				/* In case of out-of-memory error, break glass
-				$passwordfile = fopen($filename, "r");
-				while (!feof($passwordfile)) {
-					$disallowed = fgets($passwordfile);
-					if ($lowerPassword === $disallowed) {
-						$badPassword = true;
-						break;
-					}
-				}
-				fclose($passwordfile);
-				 */
-				$passwordfile = file_get_contents($filename);
-				if (strpos($passwordfile, $lowerPassword) !== false) {
+                        $sha_pass = sha1($lowerPassword);
+                        $password_hash = substr(sha1($lowerPassword),0,2);
+                        $blacklistFileSetting = $this->getSetting(CONTEXT_SITE, 'betterPasswordBlacklistFiles');
+                        if (is_null($blacklistFileSetting)) {
+                            $blacklistFileSettingHash = array();
+                            foreach ($this->getBlacklists() as $filename) {
+                                $blacklistFileSettingHash[$filename] = sha1_file($filename);
+                            }
+                            $this->updateSetting(CONTEXT_SITE, 'betterPasswordBlacklistFiles', $blacklistFileSettingHash);
+                        }
+                        else {
```

Comment:
Won't this `else` mean that if the cache creation and check will only occur if the setting is already in place?  That means the first person to touch the system gets a free pass.

Reply:
The cache creation happens outside of updating the settings. Essentially this only checks if the settings are set or not. If they aren't set, then it just sets them and the cache creation and check happen right after. If the settings are set (this else block takes care of that), the settings are updated if there are any changes to the blacklist files

Reply:
I think this will be clarified after our discussion of https://github.com/ulsdevteam/pkp-betterPassword/pull/7#discussion_r396760483